app/controllers/facilities_management/procurement_buildings_services_controller.rb

```diff
@@ -47,6 +49,36 @@ def update_procurement_building_service(pbs_params, context)
     end
   end
 
+  def update_building_area
+    @building.assign_attributes(area_params)
+
+    if building_valid?
+      @building.save
+      redirect_to facilities_management_procurement_building_path(@procurement_building)
+    else
+      params[:service_question] = params[:facilities_management_procurement_building_service][:service_question]
+      set_partial
+      render :edit
+    end
+  end
+
+  def building_valid?
+    return false unless @building.valid?(:gia)
+
+    valid = true
+    unless @procurement_building.valid?(:gia)
+      valid = false
+      @building.errors.add(:gia, :required)
+    end
+
+    unless @procurement_building.valid?(:external_area)
+      valid = false
+      @building.errors.add(:external_area, :required)
+    end
+
+    valid
```

Comment:
You can dispense with the `valid` var if you do at the end: 
```suggestion
    @building.errors.empty?
```

Reply:
Good point, I shall make that change